apps/coscrad-frontend/src/components/notes/note-detail.container.tsx

```diff
@@ -1,48 +1,37 @@
-import { useEffect } from 'react';
-import { useSelector } from 'react-redux';
-import { useAppDispatch } from '../../app/hooks';
-import { RootState } from '../../store';
-import { fetchNotes } from '../../store/slices/notes/thunks';
+import { NOT_FOUND } from '../../store/slices/interfaces/maybe-loadable.interface';
+import { useLoadableNoteById } from '../../store/slices/notes/hooks';
 import { useIdFromLocation } from '../../utils/custom-hooks/use-id-from-location';
-import { ErrorDisplay } from '../error-display/error-display';
 import { CategorizablesOfMultipleTypeContainer } from '../higher-order-components';
-import { Loading } from '../Loading';
+import { displayLoadableSearchResult } from '../higher-order-components/display-loadable-search-result';
 import { fullViewCategorizablePresenterFactory } from '../resources/factories/full-view-categorizable-presenter-factory';
 import { NoteDetailFullViewPresenter } from './note-detail.full-view.presenter';
 
 export const NoteDetailContainer = (): JSX.Element => {
-    /**
-     * TODO Once Notes follow the standard `IIndexQueryResult` structure, refactor
-     * using the same abstractions as in resource index-to-detail flows.
-     */
-    const idFromLocation = useIdFromLocation();
-
-    const dispatch = useAppDispatch();
-
-    const { data: allNotes, isLoading, errorInfo } = useSelector((state: RootState) => state.notes);
-
-    useEffect(() => {
-        if (allNotes === null) dispatch(fetchNotes());
-    }, [allNotes, dispatch]);
+    const id = useIdFromLocation();
 
-    if (errorInfo) return <ErrorDisplay {...errorInfo} />;
+    const loadableNote = useLoadableNoteById(id);
 
-    if (isLoading || allNotes === null) return <Loading />;
-
-    const note = allNotes.find(({ id }) => id === idFromLocation);
-
-    // Make this render a <NotFound />
-    if (!note) return <div>Not Found</div>;
+    const Presenter = displayLoadableSearchResult(NoteDetailFullViewPresenter);
 
+    /**
+     * TODO Now might be a good time for a helper that composes loadables.
```

Comment:
Put some thought to this.

Reply:
We already have this in our `displayLoadableSearchResult`, however, the naming obscures its very existence. The helper (and also `displayLoadableWithLoadingAndErrors`) is not so much about a container  \ presenter split as "awaiting" loaded data when a second component depends on said data to do its work.

Reply:
On further investigation, this helper had some issues. I have removed it and will leave this open for now. We really need to get a handle on chaining loadables.